### PR TITLE
[DO NOT MERGE] Cmake porting

### PR DESCRIPTION
**Pull request by rsprudencio**
Please do not merge this yet, we're setting up CMake to build openscap instead of Autotools.

### CODE REVIEW

**Comment by mpreisler** [comment]
@openscap-jenkins [skip CI]

File: config.h.in [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+#ifndef _OPENSCAP_CONFIG_H_
+#define _OPENSCAP_CONFIG_H_
+
+#define OPENSCAP_VERSION "@OPENSCAP_VERSION@"
+#define OPENSCAP_VERSION_MAJOR @OPENSCAP_VERSION_MAJOR@
+#define OPENSCAP_VERSION_MINOR @OPENSCAP_VERSION_MINOR@
+#define OPENSCAP_VERSION_PATCH @OPENSCAP_VERSION_PATCH@
+
+#define LT_CURRENT_MINUS_AGE 8
```
Comment (jan-cerny) [inline config.h.in]: This should not be hardcoded,

**Comment by jan-cerny** [comment]
Now it links successfully! 🎆 🎂 🎁 ❤️ 

```
[100%] Linking C shared library libopenscap.so
[100%] Built target openscap
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -22,4 +22,22 @@ configure_file("config.h.in" "config.h")
...
+	"${LIBXML2_INCLUDE_DIR}"
+	"${LIBXSLT_INCLUDE_DIR}"
+)
+
 add_subdirectory("src")
+
+# build "oscap" tool
+file(GLOB UTILS_SOURCES "utils/*.c")
+add_executable(oscap ${UTILS_SOURCES})
+target_link_libraries(oscap openscap ${LIBXML_LIBRARIES} ${LIBXSLT_LIBRARIES} ${LIBXSLT_EXSLT_LIBRARIES} ${PCRE_LIBRARIES} ${CURL_LIBRARIES})
```
Comment (mpreisler) [inline CMakeLists.txt]: I wish this was in the `utils` subdir. Let's keep the root CMakeLists.txt reasonably small.

File: src/OVAL/fts_sun.c [hunk 3]
```diff
@@ -26,6 +26,7 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+#if defined (__SVR4) && defined (__sun)
```
Comment (mpreisler) [inline src/OVAL/fts_sun.c]: ugh, shouldn't we be doing this in the build system?

File: CMakeLists.txt [hunk 4]
```diff
@@ -22,4 +22,22 @@ configure_file("config.h.in" "config.h")
...
+	"src/source/public"
+	"src/OVAL/public"
+	"src/XCCDF/public"
+	"src/XCCDF_POLICY/public"
+	"src/CVSS/public"
+	"src/CPE/public"
+	"src/CVE/public"
+	"${LIBXML2_INCLUDE_DIR}"
+	"${LIBXSLT_INCLUDE_DIR}"
+)
```
Comment (mpreisler) [inline CMakeLists.txt]: This has side effects! We need to move all of this in utils/CMakeLists.txt and then do add_subdirectory("utils").

**Comment by mpreisler** [comment]
@WesleyCeraso 

Cool stuff!

My .2c:
1) please make smaller commits, only one change at a time, this is really hard to review
2) we decided early on to avoid static libs and the whole linking shenanigans we do in maint-1.2, instead we want one shared library and easier builds
3) we definitely need a cmake macro or function for the probes, hardcoding and copying code around will be unmaintainable

**Comment by WesleyCeraso** [comment]
@mpreisler

Thank you for your review.

I find it difficult to commit small changes when porting to cmake because of the dependencies, one thing leads to another and at the end I have several changes, but I promisse I'll try harder with the next commits.
For points 2 & 3, my first goal it to imitate the configure.ac.tpl as much as possible, so that it generates the makefiles and builds as close as possible to the current version.
Once it is working we can start changing the cmakelists, there is much room for improvement.

File: src/OVAL/oval_filter.c [hunk 5]
```diff
@@ -35,8 +35,8 @@
 #include <string.h>
 #include <stdlib.h>
 
-#include "common/debug_priv.h"
-#include "common/elements.h"
+#include <common/debug_priv.h>
+#include <common/elements.h>
```
Comment (mpreisler) [inline src/OVAL/oval_filter.c]: the convention is to use "" if you are including from openscap and <> if you are including from system.

File: tests/API/XCCDF/unittests/unit_helper.h [hunk 6]
```diff
@@ -23,8 +23,8 @@
 #ifndef UNIT_HELPER_H_
 #define UNIT_HELPER_H_
 
-#include <xccdf_benchmark.h>
-#include <xccdf_policy.h>
+#include <XCCDF/xccdf_benchmark.h>
+#include <XCCDF_POLICY/xccdf_policy.h>
```
Comment (mpreisler) [inline tests/API/XCCDF/unittests/unit_helper.h]: this is very suspicious. if the include paths are setup correctly this shouldn't be necessary.

File: src/OVAL/probes/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,167 @@
+# PThread
+find_package(Threads REQUIRED)
+set(CMAKE_THREAD_PREFER_PTHREAD)
```
Comment (mpreisler) [inline src/OVAL/probes/CMakeLists.txt]: we should do this in the root CMakeLists.txt I think

**Comment by jan-cerny** [comment]
I still have a lot of errors in build/CMakeFiles/CMakeOutput.log

**Comment by jan-cerny** [comment]
I think this should be closed because work continues in #890

**Pull request closed without merging**